### PR TITLE
reorganizing restore code more and small tweaks

### PR DESCRIPTION
**Pull request by czue**
wait for tests. can be read commit-by-commit. 95%+ just moving stuff around.

@snopoke cc @emord

### CODE REVIEW

File: corehq/ex-submodules/casexml/apps/phone/data_providers/case/batched.py [hunk 1]
```diff
@@ -0,0 +1,543 @@
...
+    The record of how a case should sync
+    """
+    def __init__(self, case, sync_token, required_updates=None):
+        self.case = case
+        self.sync_token = sync_token
+        # cache this property since computing it can be expensive
+        self.required_updates = required_updates if required_updates is not None else self._get_required_updates()
+
+
+    def _get_required_updates(self):
```
Comment (dimagimon) [inline corehq/ex-submodules/casexml/apps/phone/data_providers/case/batched.py]: E303 too many blank lines (2)

File: corehq/ex-submodules/casexml/apps/phone/data_providers/case/providers.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+        case_response, num_batches = get_case_payload_batched(
+            restore_state=restore_state,
+            stock_settings=stock_settings,
+        )
+        # keep track of the number of batches (if any) for comparison in unit tests
+        restore_state.provider_log['num_case_batches'] = num_batches
+        return case_response
+
+
```
Comment (dimagimon) [inline corehq/ex-submodules/casexml/apps/phone/data_providers/case/providers.py]: W391 blank line at end of file

File: corehq/ex-submodules/casexml/apps/phone/restore.py [hunk 3]
```diff
@@ -571,12 +446,9 @@ def get_payload(self):
         if cached_response.exists():
             return cached_response
 
-        user = self.user
-        # create a sync log for this
         self.restore_state.start_sync()
 
-        # start with standard response
-        with self.restore_state.restore_class(user.username, items=self.params.include_item_count) as response:
+        with self.restore_state.restore_class(self.user.username, items=self.params.include_item_count) as response:
```
Comment (dimagimon) [inline corehq/ex-submodules/casexml/apps/phone/restore.py]: E501 line too long (116 > 115 characters)

**Comment by snopoke** [comment]
Looks good

**Comment by czue** [comment]
:white_check_mark:

**Pull request merged**